controllers/policystatus_utils.go

```diff
@@ -80,14 +80,23 @@ func SetPolicyConfigurationCondition(policyServerConfigMap *corev1.ConfigMap, po
 	}
 }
 
+// Checks if the given replica is owned by the given policy server deployment
+// and if it is the latest replica set from the deployment as well.
+func isReplicaSetFromPolicyServerDeployment(replicaSet *appsv1.ReplicaSet, policyServerDeployment *appsv1.Deployment) bool {
+	return replicaSet.Labels[constants.PolicyServerLabelKey] == policyServerDeployment.Labels[constants.PolicyServerLabelKey] &&
+		replicaSet.Annotations[constants.KubernetesRevisionAnnotation] == policyServerDeployment.Annotations[constants.KubernetesRevisionAnnotation] &&
+		replicaSet.Annotations[constants.PolicyServerDeploymentConfigVersionAnnotation] == policyServerDeployment.Annotations[constants.PolicyServerDeploymentConfigVersionAnnotation]
+}
+
```

Comment:
For this check I think we can just filter by the policy server label, given we are checking "ownership". So, something like:

```
func isReplicaSetFromPolicyServerDeployment(replicaSet *appsv1.ReplicaSet, policyServerDeployment *appsv1.Deployment) bool {
	return replicaSet.Labels[constants.PolicyServerLabelKey] == policyServerDeployment.Labels[constants.PolicyServerLabelKey]
}
```

Should be enough.

Reply:
But we can have more than one replica set for the same deployment. Kubernetes keep a history of the replicaset for some iterations. Do the list method remove those ones?  

https://kubernetes.io/docs/concepts/workloads/controllers/deployment/#revision-history-limit

Reply:
Never mind, the function does not need to find the latest `replicaset` in the end.